src/analysis/utils/cost-utils.ts

```diff
@@ -118,6 +118,7 @@ export const runtimeConfigCost = (config: GoogleRuntimeConfig): number => {
   if (!config) return 0;
   const computeRegion = config.normalizedRegion;
 
+  // eslint-disable-next-line no-nested-ternary
   const machineType: string = isGceRuntimeConfig(config)
```

Comment:
yeah, I think I need convincing on why adding this rule is a good thing... seems like the benefit of type narrowing is a common enough need... which in the current trend would become more common as cond usage is switched over to native for that type narrowing benefit...

Reply:
The motivation for this PR was @cjllanwarne's observation that, if we're already rejecting nested ternaries at the PR review step, it would be preferable to shift that left and into ESLint (https://github.com/DataBiosphere/terra-ui/pull/4652#discussion_r1489916173). That tightens the feedback loop.

As for why we're rejecting nested ternaries, I think they're pretty widely (though not universally held) to be more difficult to read than the alternatives. The ESLint rule's description says:
> Nesting ternary expressions can make code more difficult to understand.
> --https://eslint.org/docs/latest/rules/no-nested-ternary

Also, a while back we adopted AirBnB's ESLint configuration based on it being a widely used standard, and assuming that a widely used standard is preferable to a custom configuration. Disabling this rule is one place Terra UI differs from that standard. Applying the logic from the Lodash/FP discussion... do we have a compelling reason to differ here? Nested ternaries aren't frequently used in Terra UI, and where they are, there are alternatives like are likely easier to understand.
https://github.com/airbnb/javascript?tab=readme-ov-file#comparison--nested-ternaries

As for the type narrowing in `cond`, I don't find that particularly relevant. That's a strike against `cond`, not a point in favor of _nested_ ternaries. In a code base without `cond`, I would likely still suggest enabling this rule. As-is, it looks like only _one, maybe two_, existing uses of nested ternaries take advantage of type narrowing. And there again, if a nested ternary is the best tool for the job, we can add an ignore comment and use it. But enabling the ESLint rule points out that it's often not the best tool for the job and nudges developers to look for an alternative.

Reply:
This is the case where a nested ternary takes advantage of type narrowing. I would argue that a better way to write this would be to extract a function like `getMachineTypeFromComputeConfig`. That function be written using `if/else`, which would still take advantage of type narrowing and would be unit testable outside the component.